python/pyspark/ml/feature.py

```diff
@@ -5072,7 +5073,7 @@ def __init__(
         self._setDefault(
             stopWords=StopWordsRemover.loadDefaultStopWords("english"),
             caseSensitive=False,
-            locale=self._java_obj.getLocale(),
+            locale="en_US" if isinstance(self._java_obj, str) else self._java_obj.getLocale(),
```

Comment:
PySpark Classic calls the jvm side to get the default value
https://github.com/apache/spark/blob/e7e082663b94d0cc4d8d3d224bd8d819c4b2f4d3/mllib/src/main/scala/org/apache/spark/ml/feature/StopWordsRemover.scala#L129-L138

which is not available in Connect Mode.

Reply:
let me rethink here, if we skip _setDefault locale, then `remover.getLocale()` fails if user does not explicitly set it, but it will reuse the default value in the jvm side.